web/src/app/webview/src/keymanEngine.ts

```diff
@@ -1,31 +1,46 @@
+import { DeviceSpec } from '@keymanapp/keyboard-processor'
 import { KeymanEngine as KeymanEngineBase } from 'keyman/engine/main';
-import { Configuration } from "keyman/engine/configuration";
 import { AnchoredOSKView, ViewConfiguration, StaticActivator } from 'keyman/engine/osk';
+import { toPrefixedKeyboardId, toUnprefixedKeyboardId } from 'keyman/engine/keyboard-cache';
 
+import { WebviewConfiguration, WebviewInitOptionDefaults, WebviewInitOptionSpec } from './configuration.js';
 import ContextManager from './contextManager.js';
 import PassthroughKeyboard from './passthroughKeyboard.js';
 import { buildEmbeddedGestureConfig, setupEmbeddedListeners } from './oskConfiguration.js';
 
 export class KeymanEngine extends KeymanEngineBase<ContextManager, PassthroughKeyboard> {
-  constructor(config: Configuration, worker: Worker) {
-    // TODO:  set the old `namespacedID` function in a new Configuration property
-    // for use within `KeyboardInterface.registerStub` when available.
-    //
-    // Or... just build it in and configure via boolean flag?
-    super(config, worker, new ContextManager());
+  // Ideally, we would be able to auto-detect `sourceUri`: https://stackoverflow.com/a/60244278.
+  // But it's too new of a feature to utilize... and also expects to be in a module, when this may
+  // be compiled down to an IIFE.
+  constructor(worker: Worker, sourceUri: string) {
+    const config = new WebviewConfiguration(sourceUri);  // currently set to perform device auto-detect.
+    config.stubNamespacer = (stub) => {
+      // If the package has not yet been applied as namespacing...
+      if(stub.KP && stub.KI.indexOf(`${stub.KP}::`) == -1) {
```

Comment:
Couldn't you just search for `::`? Although wouldn't it be better for the namespace to be separated from `KI` rather than mutating input data?

Reply:
> Although wouldn't it be better for the namespace to be separated from KI rather than mutating input data?

I could see an argument for that, but the pre-modular namespacing system worked by prepending the package name as part of the ID.  I opted to maintain that.

There's then the issue of `Package_id::Keyboard_kbd_id` vs `Keyboard_Package_id::kbd_id` due to the whole `Keyboard_` ID-prefixing prepend bit that's been in KMW since forever.  The function highlighted in the code block above is designed to ensure that it's always the former pattern and never the latter by detecting the latter and converting it over while leaving the former intact.

Though... I'm not 100% sure that Package-prepended keyboards actually use the `Keyboard_` bit too in the existing implementation.